### ISSUE TITLE
Cannot be built with GOPROXY enabled, invalid entry in go.mod

### ISSUE THREAD

**Issue by fd0** [cloudflare/cloudflare-go#335]
## Expected behavior

Building with `GOPROXY=https://proxy.golang.org` enabled works.

## Current behavior

Building fails:

```
% git describe --tags
v0.9.4-8-gcc39efb

% go version
go version go1.12.7 linux/amd64

% GOPROXY=https://proxy.golang.org go build
go: finding golang.org/x/time v0.0.0-20190513212739-9d24e82272b4
go: golang.org/x/time@v0.0.0-20190513212739-9d24e82272b4: unexpected status (https://proxy.golang.org/golang.org/x/time/@v/v0.0.0-20190513212739-9d24e82272b4.info): 410 Gone
go: error loading module requirements
```

The timestamp in the pseudo-version is wrong, the referenced commit has a different timestamp:

```
% curl -i https://proxy.golang.org/golang.org/x/time/@v/v0.0.0-20190513212739-9d24e82272b4.info
HTTP/2 410 
[...]

not found: golang.org/x/time@v0.0.0-20190513212739-9d24e82272b4: invalid pseudo-version: does not match version-control timestamp (2019-03-08T20:28:27Z)
```

## Possible solution

 * Remove the timestamp from `go.mod`:
```diff
diff --git a/go.mod b/go.mod
index 58bcadc..e137c8d 100644
--- a/go.mod
+++ b/go.mod
@@ -7,5 +7,5 @@ require (
        github.com/pkg/errors v0.8.1
        github.com/stretchr/testify v1.3.0
        github.com/urfave/cli v1.21.0
-       golang.org/x/time v0.0.0-20190513212739-9d24e82272b4
+       golang.org/x/time 9d24e82272b4
 )
```
 * run `go mod tidy`, which inserts the correct timestamps:
```diff
diff --git a/go.mod b/go.mod
index 58bcadc..46fd74f 100644
--- a/go.mod
+++ b/go.mod
@@ -7,5 +7,5 @@ require (
        github.com/pkg/errors v0.8.1
        github.com/stretchr/testify v1.3.0
        github.com/urfave/cli v1.21.0
-       golang.org/x/time v0.0.0-20190513212739-9d24e82272b4
+       golang.org/x/time v0.0.0-20190308202827-9d24e82272b4
 )
```
 * release new version with fixed `go.mod`

## Context

I noticed that I'm not able to build [lego](https://github.com/go-acme/lego) using the new `GOPROXY`:

```
% go version
go version go1.12.7 linux/amd64

% export GOPATH=$(mktemp -d)
% export GOPROXY=https://proxy.golang.org

% git clone https://github.com/go-acme/lego
[...]

% cd lego/cmd/lego
% go build
[...]
go: golang.org/x/lint@v0.0.0-20190511005446-959b441ac422: unexpected status (https://proxy.golang.org/golang.org/x/lint/@v/v0.0.0-20190511005446-959b441ac422.info): 410 Gone
[...]
go: golang.org/x/time@v0.0.0-20190513212739-9d24e82272b4: unexpected status (https://proxy.golang.org/golang.org/x/time/@v/v0.0.0-20190513212739-9d24e82272b4.info): 410 Gone
[...]
go: error loading module requirements
```

It builds without the proxy, probably because the proxy now does stricter [version validation](https://tip.golang.org/doc/go1.13#version-validation), which will be part of Go 1.13:

Both errors can be traced back to invalid entries in `go.mod` of `github.com/cloudflare/cloudflare-go` in version `v0.9.4`:

```
	golang.org/x/lint v0.0.0-20190511005446-959b441ac422 // indirect
	golang.org/x/time v0.0.0-20190513212739-9d24e82272b4
```

I'm using `golang.org/x/time` as an example now because it's still in `go.mod` in the `master` branch.

I was very curious where the time stamp came from, and it looks like a bot (which probably needs to be fixed) inserted it in 4922c512a7435822eb6584d7c2642c5aa3baba2e:

```diff
commit 4922c512a7435822eb6584d7c2642c5aa3baba2e
Author: Renovate Bot <bot@renovateapp.com>
Date:   Mon May 13 23:47:33 2019 +0200

    Update golang.org/x/time commit hash to 9d24e82 (#304)

diff --git a/go.mod b/go.mod
index d887e16..71076e3 100644
--- a/go.mod
+++ b/go.mod
@@ -8,5 +8,5 @@ require (
        github.com/pkg/errors v0.8.0
        github.com/stretchr/testify v1.3.0
        golang.org/x/lint v0.0.0-20190511005446-959b441ac422 // indirect
-       golang.org/x/time v0.0.0-20180412165947-fbb02b2291d2
+       golang.org/x/time v0.0.0-20190513212739-9d24e82272b4
 )
```

The [commit metadata](https://github.com/golang/time/commit/9d24e82272b4f38b78bc8cff74fa936d31ccd8ef) does not mention the timestamp anywhere:
```
commit 9d24e82272b4f38b78bc8cff74fa936d31ccd8ef (HEAD -> master, origin/master, origin/HEAD)
Author:     Michael Matloob <matloob@golang.org>
AuthorDate: Fri Feb 15 17:48:40 2019 -0500
Commit:     Michael Matloob <matloob@golang.org>
CommitDate: Fri Mar 8 20:28:27 2019 +0000
[...]
```

## Your Environment

* cc39efbdaa777f0dada061fe6eb2140c3735f556, v0.9.4
* Go 1.12.7, Go 1.13beta1

**Comment by ldez** [comment]
I already open a PR to fix that: #334

You can also see #333

**Comment by fd0** [comment]
Dang, I didn't check the pull requests and spent time digging into the problem. Sigh. Thanks for the pointers!
